lua/tardis/screens/cl_screen_coordinates.lua

```diff
@@ -45,136 +40,116 @@ TARDIS:AddScreen("Destination", {id="coordinates", text="Screens.Coordinates", m
-            else
-                self:SetText(math.Round(ext:GetPos().z, round_digits))
-            end
-        end
-    end
 
-    local pos_pitch = vgui.Create("DTextEntry",PositionPanel)
+    local pos_pitch = vgui.Create("DTextEntry",position_panel)
     pos_pitch:SetPlaceholderText("")
     pos_pitch:SetPos(panel_d, 2 * pos_elem_h + 3 * panel_d)
     pos_pitch:SetSize(pos_elem_w,pos_elem_h)
     pos_pitch:SetNumeric(true)
     pos_pitch:SetEnabled(false)
     pos_pitch:SetFont(font)
-    pos_pitch.Think = function(self)
-        if IsValid(ext) then
-            self:SetText(math.Round(ext:GetAngles().p, round_digits))
-        end
-    end
 
-    local pos_yaw = vgui.Create("DTextEntry",PositionPanel)
+    local pos_yaw = vgui.Create("DTextEntry",position_panel)
     pos_yaw:SetPlaceholderText("")
     pos_yaw:SetPos(pos_elem_w + 2 * panel_d, 2 * pos_elem_h + 3 * panel_d)
     pos_yaw:SetSize(pos_elem_w,pos_elem_h)
     pos_yaw:SetNumeric(true)
     pos_yaw:SetEnabled(false)
     pos_yaw:SetFont(font)
-    pos_yaw.Think = function(self)
-        if IsValid(ext) then
-            self:SetText(math.Round(ext:GetAngles().y, round_digits))
-        end
-    end
 
-    local pos_roll = vgui.Create("DTextEntry",PositionPanel)
+    local pos_roll = vgui.Create("DTextEntry",position_panel)
     pos_roll:SetPlaceholderText("")
     pos_roll:SetPos(pos_elem_w * 2 + 3 * panel_d, 2 * pos_elem_h + 3 * panel_d)
     pos_roll:SetSize(pos_elem_w,pos_elem_h)
     pos_roll:SetNumeric(true)
     pos_roll:SetEnabled(false)
     pos_roll:SetFont(font)
-    pos_roll.Think = function(self)
-        if IsValid(ext) then
-            self:SetText(math.Round(ext:GetAngles().r, round_digits))
-        end
-    end
 
-    local pos_save = vgui.Create("DButton", PositionPanel)
+    local pos_save = vgui.Create("DButton", position_panel)
     pos_save:SetSize(pos_elem_w,pos_elem_h)
     pos_save:SetPos(pos_elem_w * 3 + 4 * panel_d, pos_elem_h + 2 * panel_d)
     pos_save:SetText(TARDIS:GetPhrase("Screens.Coordinates.Save"))
     pos_save:SetFont(font)
-    pos_save.Think = function(self)
-        if not IsValid(ext) then return end
-        EnsureEnabled(self, not ext:GetData("vortex"))
-    end
 
-    local pos_copy = vgui.Create("DButton", PositionPanel)
+    local pos_copy = vgui.Create("DButton", position_panel)
     pos_copy:SetSize(pos_elem_w,pos_elem_h)
     pos_copy:SetPos(pos_elem_w * 3 + 4 * panel_d, 2 * pos_elem_h + 3 * panel_d)
     pos_copy:SetText(TARDIS:GetPhrase("Screens.Coordinates.Copy"))
     pos_copy:SetFont(font)
-    pos_copy.Think = function(self)
+
+    local function generate_vortex_coordinate()
+        if not TARDIS:GetSetting("gui_animations") then
+            return "???"
+        end
+        return math.random(-99999999, 99999999) * 0.0001
+    end
+
+    position_panel.Think = function(self)
         if not IsValid(ext) then return end
-        EnsureEnabled(self, not ext:GetData("vortex"))
+        if ext:GetData("vortex") then
```

Comment:
We are calling `ext:GetData("vortex")` in this function 3 times, cache it in a local variable and re-use

Reply:
fixed